plugins/modules/pfsense_virtual_ip.py

```diff
@@ -0,0 +1,316 @@
+from ansible_collections.pfsensible.core.plugins.module_utils.module_base import PFSenseModuleBase
+
+
+VIRTUALIP_ARGUMENT_SPEC = dict(
+    mode=dict(required=True, choices=['proxyarp', 'carp', 'ipalias', 'other'], type='str'),
+    interface=dict(required=True, type='str'),
+    vhid=dict(required=False, type='int'),
+    advskew=dict(required=False, type='int'),
+    advbase=dict(required=False, type='int'),
+    password=dict(required=False, type='str', no_log=True),
+    uniqid=dict(required=False, type='str'),
+    descr=dict(required=False, type='str'),
+    type=dict(required=False, type='str', choices=['single'], default='single'),
+    subnet_bits=dict(required=False, type='int', default=32),
+    subnet=dict(required=False, type='str'),
+    state=dict(default='present', choices=['present', 'absent'], type='str'),
+)
+
+VIRTUALIP_REQUIRED_IF = [
+    ["mode", "carp", ["uniqid", "password", "vhid", "advbase"]],
+    ["mode", "ipalias", ["uniqid"]],
+]
+
+# map field names between ansible and pfsense
+params_map = {}
+
+# fields that are not written to pfsense
+skip_list = ['state']
+
+
+class PFSenseVirtualIPModule(PFSenseModuleBase):
+    """ module managing pfsense virtual ip settings """
+
+    @staticmethod
+    def get_argument_spec():
+        """ return argument spec """
+        return VIRTUALIP_ARGUMENT_SPEC
+
+    ##############################
+    # init
+    #
+    def __init__(self, module, pfsense=None):
+        super(PFSenseVirtualIPModule, self).__init__(module, pfsense)
+        self.name = "virtual_ip"
+        self.root_elt = self.pfsense.get_element('virtualip')
+        self.obj = dict()
+
+        if self.root_elt is None:
+            self.root_elt = self.pfsense.new_element('virtualip')
+            self.pfsense.root.append(self.root_elt)
+
+    ##############################
+    # params processing
+    #
+    def _params_to_obj(self):
+        """ return a dict from module params """
+        params = self.params
+
+        obj = dict()
+        self.obj = obj
+
+        def _set_param(target, param):
+            if params.get(param) is not None:
+                if isinstance(params[param], str):
+                    target[param] = params[param]
+                else:
+                    target[param] = str(params[param])
+
+        def _set_param_bool(target, param):
+            if params.get(param) is not None:
+                value = params.get(param)
+                if value is True and param not in target:
+                    target[param] = ''
+                elif value is False and param in target:
+                    del target[param]
+
+        for param in VIRTUALIP_ARGUMENT_SPEC:
+            if param not in skip_list:
+                if VIRTUALIP_ARGUMENT_SPEC[param]['type'] == 'bool':
+                    _set_param_bool(obj, param)
+                else:
+                    _set_param(obj, param)
+
+        return obj
+
+    def _validate_params(self):
```

Comment:
Is there no validation to be done?

Reply:
We do not have some, should we delete that function (replace with `pass`) and hope that somebody else write some?

Or is it an blocker?